test/functional/terminal/highlight_spec.lua

```diff
@@ -134,7 +134,7 @@ describe('terminal window highlighting with custom palette', function()
     })
     screen:attach(true)
     nvim('set_var', 'terminal_color_3', '#123456')
-    execute('enew | call termopen(["'..nvim_dir..'/tty-test"]) | startinsert')
+    execute('enew | let $NVIM_TUI_ENABLE_TRUE_COLOR=1 | call termopen(["'..nvim_dir..'/tty-test"]) | startinsert')
```

Comment:
This should not be necessary; in the `screen:attach(true)` call above, `true` specifies that the attached UI is an RGB ("true color") UI. That's what you should check in `terminal_open()`, not `os_getenv("NVIM_TUI_ENABLE_TRUE_COLOR") != NULL`